astropy/coordinates/angle_utilities.py

```diff
@@ -184,33 +202,62 @@ def p_colon(p):
             '''
             colon : sign UINT COLON ufloat
                   | sign UINT COLON UINT COLON ufloat
+                  | sign UINT COLON ufloat dir
+                  | sign UINT COLON UINT COLON ufloat dir
             '''
+            if p.slice[len(p) - 1].type == 'dir':
+                if p[1] == -1:
+                    raise ValueError()
             if len(p) == 5:
                 p[0] = (p[1] * p[2], p[4])
+            elif len(p) == 6:
```

Comment:
With my suggestion above, you now do need to check the type of `p[0]` for being 'sign' or of `p[-1]` (does that work??) for being 'dir'

Reply:
no p[-1] is not possible and sign is always present no matter what because in the grammer it is defined that if no sign is present then assume it to be 1.0.